File: client/src/types/product.ts
export type Product = {
  id: number;
  sku: string;
  name: string;
  description: string;
  price: number;
  isActive: boolean;
  categoryId: number;
  categoryName: string;
  stockOnHand: number;
  stockReserved: number;
  reorderThreshold: number;
  createdAtUtc: string;
};

export type Category = {
  id: number;
  name: string;
};

export type CreateProductRequest = {
  sku: string;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  stockOnHand: number;
  stockReserved: number;
  reorderThreshold: number;
};

File: Domain/Entities/Product.cs
namespace Domain.Entities;

public class Product
{
    public int Id { get; set; }
    public string Sku { get; set; } = null!;
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public bool IsActive { get; set; } = true;
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public InventoryItem? InventoryItem { get; set; }
    public ICollection<InventoryMovement> InventoryMovements { get; set; } = new List<InventoryMovement>();
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAtUtc { get; set; }
}
